src/sniffles/sniffles.py

```diff
@@ -183,14 +183,19 @@ def start_generation(sconf):
     # for recording how packets are generated
     fd_result = open(sconf.getResultFile(), 'w')
 
+    if allrules:
+        i = random.randrange(len(allrules))
+        
     while current < end:
-        myrule = None
+        if i == len(allrules):
```

Comment:
I believe you can update the cursor in a single line.  Something like:
rule_cursor = (rule_cursor + 1) % len(all_rules) if all_rules else 0

That would remove the need to explicitly increment rule_cursor (i).